toonz/sources/toonz/startuppopup.cpp

```diff
@@ -0,0 +1,889 @@
+  RecentFiles::instance()->refreshRecentFilesMenu(RecentFiles::Scene);
+  hide();
+}
+
+//-----------------------------------------------------------------------------
+
+void StartupPopup::onCameraUnitChanged(int index) {
+  Preferences *pref = Preferences::instance();
+  QStringList type;
+  type << tr("pixel") << tr("cm") << tr("mm") << tr("inch") << tr("field");
+
+  double width  = m_widthFld->getValue();
+  double height = m_heightFld->getValue();
+  if (index != 0) {
+    pref->setPixelsOnly(false);
+    pref->setCameraUnits(type[index].toStdString());
+    m_widthFld->setDecimals(4);
+    m_heightFld->setDecimals(4);
+    m_resTextLabel->show();
+    m_resXFld->show();
+    m_resYFld->show();
+    m_resXLabel->show();
+    m_dpiFld->show();
+    m_dpiLabel->show();
+    m_widthFld->setMeasure("camera.lx");
+    m_heightFld->setMeasure("camera.ly");
+    m_widthFld->setValue(width);
+    m_heightFld->setValue(height);
+  } else {
+    pref->setPixelsOnly(true);
+    pref->setUnits("pixel");
+    pref->setCameraUnits("pixel");
+    m_widthFld->setDecimals(0);
+    m_heightFld->setDecimals(0);
+    m_resTextLabel->hide();
+    m_resXFld->hide();
+    m_resYFld->hide();
+    m_resXLabel->hide();
+    m_dpiFld->hide();
+    m_dpiLabel->hide();
+    m_dpiFld->setValue(Stage::standardDpi);
+    m_widthFld->setMeasure("camera.lx");
+    m_heightFld->setMeasure("camera.ly");
+    m_widthFld->setValue(m_xRes / Stage::standardDpi);
+    m_heightFld->setValue(m_yRes / Stage::standardDpi);
+  }
+}
+
+//-----------------------------------------------------------------------------
+
+void StartupPopup::onShowAtStartChanged(int index) {
+  Preferences::instance()->enableStartupPopup(index);
+}
+
+//-----------------------------------------------------------------------------
+
+void StartupPopup::updateResolution() {
+  if (Preferences::instance()->getPixelsOnly()) {
+    if (m_dpiFld->getValue() != Stage::standardDpi) {
+      m_dpiFld->setValue(Stage::standardDpi);
+    }
+    m_xRes = m_widthFld->getValue() * Stage::standardDpi;
+    m_yRes = m_heightFld->getValue() * Stage::standardDpi;
+    m_resXFld->setValue(m_xRes);
+    m_resYFld->setValue(m_yRes);
+  } else {
+    m_xRes = m_widthFld->getValue() * m_dpi;
+    m_yRes = m_heightFld->getValue() * m_dpi;
+    m_resXFld->setValue(m_xRes);
+    m_resYFld->setValue(m_yRes);
+  }
+  m_presetCombo->setCurrentIndex(0);
+}
+
+//-----------------------------------------------------------------------------
+
+void StartupPopup::updateSize() {
+  m_xRes = m_resXFld->getValue();
+  m_yRes = m_resYFld->getValue();
+  if (Preferences::instance()->getPixelsOnly()) {
+    if (m_dpiFld->getValue() != Stage::standardDpi) {
+      m_dpiFld->setValue(Stage::standardDpi);
+    }
+    m_widthFld->setValue(m_xRes / Stage::standardDpi);
+    m_heightFld->setValue(m_yRes / Stage::standardDpi);
+  } else {
+    m_widthFld->setValue(m_xRes / m_dpi);
+    m_heightFld->setValue(m_yRes / m_dpi);
+  }
```

Comment:
For now, inputting pixel size field causes unexpected change in width/height fields - it looses fractional portion.
Lines 865-869 should be ;
`    m_widthFld->setValue((double)m_xRes / Stage::standardDpi);`
`    m_heightFld->setValue((double)m_yRes / Stage::standardDpi);`
`  } else {`
`    m_widthFld->setValue((double)m_xRes / (double)m_dpi);`
`    m_heightFld->setValue((double)m_yRes / (double)m_dpi);`